### DISCUSSION TITLE
Reset password with express

### DISCUSSION THREAD

**Discussion by sumoncse19** [clerk/javascript#4464] [Support request - DO NOT OPEN HERE]
I want to try to implement reset password of clerk with express, but it's not working.

`
async resetPasswordRequest(email: string) {
    try {
      const user = await this.userRepository.findOne({ where: { email } })
      if (!user) {
        throw new AppError(404, 'User not found')
      }

      // Create sign-in attempt with reset password strategy
      const signInAttempt = await clerk.signIns.attemptFirstFactor({
        identifier: email,
        strategy: 'reset_password_email_code',
      })

      if (!signInAttempt) {
        throw new AppError(500, 'Failed to initiate password reset')
      }

      // Prepare first factor verification to send the code
      await signInAttempt.prepareFirstFactor({
        strategy: 'reset_password_email_code',
      })

      return {
        success: true,
        message: 'Password reset verification code has been sent to your email',
        signInId: signInAttempt.id,
      }
    } catch (error) {
      console.error('Reset password request error:', error)
      if (error instanceof AppError) {
        throw error
      }
      throw new AppError(500, 'Failed to initiate password reset')
    }
  }

  async resetPassword(code: string, newPassword: string) {
    try {
      // Create a sign-in attempt
      const signInAttempt = await clerk.signIns.create({
        strategy: 'reset_password_email_code',
      })

      // Attempt first factor verification with the code and new password
      const result = await signInAttempt.attemptFirstFactor({
        strategy: 'reset_password_email_code',
        code,
        password: newPassword,
      })

      // Check if 2FA is required
      if (result.status === 'needs_second_factor') {
        throw new AppError(
          400,
          '2FA is required but not supported in this flow',
        )
      }

      // Check if the reset was successful
      if (result.status !== 'complete') {
        throw new AppError(400, 'Password reset failed')
      }

      return {
        success: true,
        message: 'Password has been reset successfully',
        sessionId: result.createdSessionId,
      }
    } catch (error) {
      console.error('Reset password error:', error)
      if (error instanceof AppError) {
        throw error
      }

      // Handle specific Clerk errors
      if (error && typeof error === 'object' && 'errors' in error) {
        const clerkError = error as {
          errors: Array<{
            code: string
            message: string
            longMessage?: string
          }>
        }

        if (clerkError.errors[0]?.code === 'form_password_pwned') {
          throw new AppError(
            422,
            'This password has been compromised in a data breach. Please choose a different password.',
          )
        }

        throw new AppError(
          400,
          clerkError.errors[0]?.longMessage ||
            clerkError.errors[0]?.message ||
            'Invalid verification code or password',
        )
      }

      throw new AppError(400, 'Invalid verification code or password')
    }
  }
`

and this is my clerk.ts file code:
`
import { createClerkClient } from '@clerk/backend'
import { clerkMiddleware, requireAuth } from '@clerk/express'
import dotenv from 'dotenv'

dotenv.config()

if (!process.env.CLERK_SECRET_KEY) {
  throw new Error('CLERK_SECRET_KEY is not set in environment variables')
}

// Initialize Clerk client for API operations
export const clerk = createClerkClient({
  secretKey: process.env.CLERK_SECRET_KEY,
})

// Export middleware
export const clerkAuth = clerkMiddleware()
export { requireAuth }
`

I tried a lot's of way but i couldn't find any solution.

**Comment by jescalan** [comment]
Please open this issue with [clerk support](https://clerk.com/contact/support) instead!
